shared/src/main/kotlin/org/jetbrains/kotlin/konan/target/Linker.kt

```diff
@@ -265,7 +265,7 @@ class MacOSBasedLinker(targetProperties: AppleConfigurables)
         if (debug) {
             result += dsymUtilCommand(executable, outputDsymBundle)
             if (optimize) {
-                result += Command(strip, "-S", executable)
+                result += Command(strip, linkerNoDebugFlags, executable)
```

Comment:
I'm not sure that sharing the same set of flags between `ld` and `strip` is a good idea. For example, changing `linkerNoDebugFlags` with something `ld`-specific will break compilation.
A better approach would be an addition of a separate `stripFlags` property. [This commit](https://github.com/JetBrains/kotlin-native/commit/6137389c757ec8ca5298b533d1044e4955d1adc2) will likely help you. There we add a single `llvmInlineThreshold` that is shared across all targets.

Reply:
I absolutely agree and will adjust it accordingly by introducing a new konan parameter to tune it.